### DISCUSSION TITLE
missing id property on p-menuitem

### DISCUSSION THREAD

**Discussion by balagex** [primefaces/community#298] [PrimeNG] [answered]
After upgrading our application to Angular 16 ("@angular/core": "^16.2.9") and PrimeNg 16 ("primeng": "^16.5.0"), I realized, id attribute of the "a" tag (with menuitem role) disappeared!

In the product environment (with Angular 13 + primeNg 13) the HTML contains the id attribute:
![kép](https://github.com/primefaces/community/assets/50229207/4897b2fd-b794-4e53-88ca-bc8e49b8f6af)

In the 16th version, the id has gone:
![kép](https://github.com/primefaces/community/assets/50229207/ea37565c-7bc4-4c22-a879-d604475b22a9)

All automated tests are failing, because those cannot click on a required menu item with specific id.

I did not found any way in the API documentation, how can I force the use if the id. If I set the automationId property of the related MenuItem class, then the "data-automationid" HTML attribute has appeared. 

But we do not want to rewrite all automated tests, so please tell, how can I get those missing id attributes back.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've made adjustments to the placement of id attributes in our menu components as part of our efforts to improve accessibility. Unfortunately, reverting these attributes to anchor tags is not feasible. However, you don't need to rewrite all of the tests. Instead of targeting anchors with the id attribute, you can now target list items, which requires only minimal changes. It's also important to note that the click functionality has been moved from the anchors to the div with `p-menuitem-content` class.
![Screenshot 2023-10-25 at 09 23 55](https://github.com/primefaces/community/assets/69278826/fc022921-2551-4400-8f26-bd22a80b51d0)

**Comment by balagex** [comment]
Hi,

Maybe I misunderstood something or simply do not see the forest behind the tree, but I don't find any id attribute on list items either. It appears in your inserted picture, but does not appear in the generated HTML in our environment.

![image](https://github.com/primefaces/community/assets/50229207/0e365778-6891-4925-b6cf-8cd66028c378)

Our menu which related to that HTML example:

![image](https://github.com/primefaces/community/assets/50229207/f3f6a198-9f41-48f9-b720-9d3c6f88aa3f)

What do I miss?

And an additional comment. I set automationId attribute of all MenuItem class, but its value appears ONLY on the top level menu items as data-automationid attribute, but not on 2nd or 3rd menu level. (see attached picture too)
